tasks/upgrade.js

```diff
@@ -0,0 +1,22 @@
+const { task } = require("hardhat/config");
+
+task("upgrade", "upgrade a contract").setAction(async () => {
+  await hre.run("compile");
+
+  if (
+    network.name !== "mumbai" &&
+    network.name !== "matic" &&
+    network.name !== "goerli"
+  ) {
+    console.log(`${network.name} not supported.`);
+    return;
+  }
+
+  const contract = await hre.ethers.getContractFactory("PosterFactory");
+
+  const address = "0xAB7D3d1a1a16D5178c283256CcA9A8A5474cf8a4";
+  implAddress = await upgrades.erc1967.getImplementationAddress(address);
+  console.log("Old implementation address:", implAddress);
+  await upgrades.upgradeProxy(address, contract);
```

Comment:
The proxy should be owned by the Mint Songs multisig, so we'll need to use defender here, which would be `proposeUpgrade`, but maybe we should extract our tasks out from the open source repo? Not sure how to go about that.

Reply:
not needed in this PR as the contract isn't currently owned by multisig.